### ISSUE TITLE
MODX 2.8.5 crach empty cart (trash manager)

### ISSUE THREAD

**Issue by rasxod** [modxcms/revolution#16429]
**resource/trash/getlist**

```
<b>Fatal error</b>:  Uncaught TypeError: Argument 1 passed to modObjectGetListProcessor::prepareQueryAfterCount() must be an instance of xPDOQuery, null given, called in D:\www\anm\core\model\modx\modprocessor.class.php on line 568 and defined in D:\www\anm\core\model\modx\modprocessor.class.php:606
Stack trace:
#0 D:\www\anm\core\model\modx\modprocessor.class.php(568): modObjectGetListProcessor-&gt;prepareQueryAfterCount()
#1 D:\www\anm\core\model\modx\modprocessor.class.php(501): modObjectGetListProcessor-&gt;getData()
#2 D:\www\anm\core\model\modx\modprocessor.class.php(185): modObjectGetListProcessor-&gt;process()
#3 D:\www\anm\core\model\modx\modx.class.php(1770): modProcessor-&gt;run()
#4 D:\www\anm\core\model\modx\modconnectorresponse.class.php(144): modX-&gt;runProcessor()
#5 D:\www\anm\core\model\modx\modconnectorrequest.class.php(86): modConnectorResponse-&gt;outputContent()
#6 D:\www\anm\core\model\modx\modconnectorrequest.class.php(73): modConnectorRequest-&gt;prepareResponse()
#7 D:\www\anm\connectors\index.php(78): modConnectorRequest-&gt; in <b>D:\www\anm\core\model\modx\modprocessor.class.php</b> on line <b>606</b><br />
```

**Comment by halftrainedharry** [comment]
Does the pull request #16415 fix the issue?

**Comment by smg6511** [comment]
Fixed with #16415 for 2.8.7+ and #16433 for 3.0.4+.
